src/EnergyPlus/DataPlant.hh

```diff
@@ -597,6 +601,7 @@ namespace DataPlant {
 		int IndexInLoopSidePumps; // If I'm a pump, this tells my index in PL(:)%LS(:)%Pumps
 		Real64 TempDesCondIn;
 		Real64 TempDesEvapOut;
+		std::shared_ptr< PlantComponent > compPtr;
```

Comment:
I'm not against this.  I didn't plan on being the first one to put raw pointers in there, but I can be.  That would certainly simplify things.  It would allow us to keep GetInput the same since it is just building up an array of instances, storage of the instances would be the same, and we wouldn't have the overhead of the shared pointer.  I think this is good.

Any objections, concerns about proceeding with this @kbenne?  And @mjwitte if you have any concerns, feel free to chime in.